### ISSUE TITLE
JAX support

### ISSUE THREAD

**Issue by crusaderky** [crusaderky/pyscenarios#35]
Blockers for JAX support, as of JAX 0.5.2:

### Sobol
`jnp.bitwise_xor.accumulate` is quite slow on CPU and _extremely_ slow on GPU. Alternative, specialized implementations exist; however not being able to query the device inside `jax.jit` means that one can't write them into `pyscenarios` without resorting to cumbersome parameters or module-level config flags.

- [ ] https://github.com/jax-ml/jax/issues/28097
- [ ] https://github.com/jax-ml/jax/issues/26000#issuecomment-2813584427

### T* copula
Missing distribution functions in [jax.scipy](https://docs.jax.dev/en/latest/jax.scipy.html):

- [ ] `jax.scipy.stats.chi2.ppf` (one could use `jax.random.chisquare` on JAX own's PRNG, but not on Sobol)
- [x] `jax.scipy.stats.t.cdf` - read below
- [x] `jax.scipy.stats.norm.ppf`
- [ ] `jax.scipy.special.betainc` is extremely slow https://github.com/jax-ml/jax/issues/28547

**Comment by crusaderky** [comment]
- `scipy.stats.chi2.ppf` is implemented by
- `scipy.stats.gamma.ppf(x, a=df/2, loc=0, scale=2)`, which is implemented by
- `2 * scipy.special.gammaincinv(df/2, x)`, which is implemented by
- `2 * cephes::igami(df/2, x)` https://github.com/scipy/xsf/blob/4fff9b2cb2b5c31a0cf0b0f609d2699a5eeac53b/include/xsf/cephes/igami.h

**Comment by crusaderky** [comment]
- XREF https://github.com/jax-ml/jax/issues/5350

**Comment by crusaderky** [comment]
`jax.scipy.t.cdf(x, df)` does not exist at the moment of writing; however it's equivalent to `scipy.special.stdtr(df, x)` in scipy 1.16 (TBC scipy 1.15). Note this is _not_ in `jax.scipy` but fully works with JAX arrays on GPU too. 

It's also trivial to backport to avoid using scipy at all.
From `scipy.special._support_alternative_backends`:
```python
def stdtr(df, t):
    x = df / (t ** 2 + df)
    tail = betainc(df / 2, 0.5, x) / 2
    return xp.where(t < 0, tail, 1 - tail)
```

**Comment by crusaderky** [comment]
Some early benchmarks:

- 65535 samples
- 1000 dimensions
- AMD Ryzen 9 3950X 16-Core
- NVidia RTX 3080
- warm JIT in all cases

| Backend | Sobol | Gaussian Copula | Student T Copula |
| --- | --: | --: | --: |
| Numpy+Numba | 0.10s | 4.89s | 27.80s |
| Dask + Numba [1] | 3.41s | 13.00s | 16.00s |
| JAX CPU | 0.18s | 0.75s | 264.00s [2] |
| JAX GPU | 0.004s | 0.38s |  10.70s [2] | 

[1] 16 chunks along the samples axis, threads scheduler
[2] `chi2.ppf` implemented by wrapping the scipy C implementation with [jax.pure_callback](https://docs.jax.dev/en/latest/_autosummary/jax.pure_callback.html#jax.pure_callback).

**Comment by crusaderky** [comment]
could [betainc](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.betainc.html) be reimplemented with [quadax](https://quadax.readthedocs.io/en/latest/_api/quadax.quadgk.html#quadax.quadgk)?
